### ISSUE TITLE
URLs to blog articles changed

### ISSUE THREAD

**Issue by benelsen** [JuliaLang/www.julialang.org#562]
After the switchover, all (most?) links to the blog articles are 404ing.

Some examples:
https://julialang.org/blog/2019/12/yao-differentiable-quantum-programming => https://julialang.org/blog/2019/12/yao.jl-differentiable-quantum-programming-in-julia/
https://julialang.org/blog/2019/11/artifacts => https://julialang.org/blog/2019/11/reliable-and-reproducible-binary-artifacts-for-julia-packages/
https://julialang.org/blog/2019/07/multithreading => https://julialang.org/blog/2019/07/announcing-composable-multi-threaded-parallelism-in-julia/

Looks like the old slugs were handwritten, the new ones just generated from the title.

**Comment by ViralBShah** [comment]
Yes, in the old blog, the blog URLs were based on the filename submitted for the blog. Now they are autogenerated from the text.

@abhishalya Can we use the filename for the URL like the old jekyll site?

**Comment by logankilpatrick** [comment]
You are referencing external URL's correct? Accessing the blog posts from the site itself seems to work

**Comment by ViralBShah** [comment]
Yes. The website is internally consistent. Because the way the blog URLs are being generated, old links will fail. So we need to do it consistently with what our jekyll site was doing, or add redirects for the old blogs.

**Comment by benelsen** [comment]
I think the preferred way would be to use the old urls.

```yaml
permalinks:
  blog: "blog/:year/:month/:slug"
```
in the config and 
```
slug: old-slug
```
in the articles should work: https://gohugo.io/content-management/urls/#permalink-configuration-values

**Comment by ViralBShah** [comment]
Could you submit a PR? I am not fully familiar with hugo, but you should be able to see the results in the CI.

**Comment by benelsen** [comment]
On it, should take 15-30 minutes

**Comment by logankilpatrick** [comment]
Merged, let's see if this reverts those changes.

**Comment by benelsen** [comment]
Someone might need to go though all of them with an [archived version](https://web.archive.org/web/20191229063745/https://julialang.org/blog/) of the site to make sure, all of them work.

**Comment by logankilpatrick** [comment]
I'm getting a 404 for https://julialang.org/blog/2019/12/yao-differentiable-quantum-programming

**Comment by logankilpatrick** [comment]
Some of the links still don't work like the one above ^

**Comment by benelsen** [comment]
I tried a few and they seem to all ork, maybe there are still some caches not refreshed yet.

**Comment by ViralBShah** [comment]
All seem to be working for me now.

**Comment by ViralBShah** [comment]
Fixed by #565

**Comment by ViralBShah** [comment]
Thanks @benelsen
